engine/core.py

```diff
@@ -134,6 +134,18 @@ def get_key_points(self, name: Optional[str] = None) -> Iterable[scripts.KeyPoin
 
         return self.world.get_key_points(name)
 
+    def get_sprites(self, name: Optional[str] = None) -> Iterable[arcade.Sprite]:
+        """Gets all sprites with the given name."""
+        if self.world is None:
+            raise GameNotInitializedError()
+
+        matching_sprites = []
+        for n in self.world.scripted_objects.items():
+            if name.startswith(n):
+                matching_sprites.append(self.world.scripted_objects[n])
+
+        return matching_sprites
```

Comment:
Python style: list comprehensions would be preferred here.

https://google.github.io/styleguide/pyguide.html#27-comprehensions--generator-expressions

```python
return [obj.sprite for obj_name, obj in self.world.scripted_objects.items() if name in obj_name]
```